frontend/app/handlers.ts

```diff
@@ -1,19 +1,47 @@
+import {
+    crypto_pwhash_str,
+    crypto_pwhash_str_verify,
+    crypto_pwhash_OPSLIMIT_INTERACTIVE,
+    crypto_pwhash_MEMLIMIT_INTERACTIVE,
+} from "libsodium-wrappers-sumo";
+
 // Make API call to server to check if password was found in breached dataset
 export const checkSecurity = async (password: string) => {
     try {
-        const response = await fetch("http://localhost:18080/intersection", {
-            method: "POST",
-            mode: "cors",
-            headers: {
-                "Access-Control-Allow-Headers": "*", // cors setting
-                "Content-Type": "application/json"
-            },
-            body: password
-        })
+        // hash the password
+        const passwordHash = crypto_pwhash_str(
+            password,
+            crypto_pwhash_OPSLIMIT_INTERACTIVE,
+            crypto_pwhash_MEMLIMIT_INTERACTIVE
+        );
+        if (!crypto_pwhash_str_verify(passwordHash, password)) {
+            console.log("unable to hash password");
+            return { status: "error" };
+        }
+
+        // TODO: encrypt 
+
+        const response = await fetch(
+            "http://localhost:18080/intersection",
+            {
+                method: "POST",
+                mode: "cors",
+                headers: {
+                    "Access-Control-Allow-Headers": "*", // cors setting
+                    "Content-Type": "application/json",
+                },
+                body: password,
```

Comment:
**question**: Would this be where we pass in the passwordHash^a in future issues?

Reply:
yes!